algorithms/linfa-preprocessing/src/count_vectorization.rs

```diff
@@ -0,0 +1,491 @@
+        );
+
+        let mut vocabulary = vocabulary
+            .into_iter()
+            .filter(|(_, (_, abs_count))| *abs_count >= min_abs_df && *abs_count <= max_abs_df)
+            .collect();
+
+        let vec_vocabulary = hashmap_to_vocabulary(&mut vocabulary);
+
+        Ok(FittedCountVectorizer {
+            vocabulary,
+            vec_vocabulary,
+            properties: self.properties.clone(),
+        })
+    }
+
+    /// Produces a [FittedCountVectorizer](struct.FittedCountVectorizer.html) with the input vocabulary.
+    /// All struct attributes are ignored in the fitting but will be used by the [FittedCountVectorizer](struct.FittedCountVectorizer.html)
+    /// to transform any text to be examined. As such this will return an error in the same cases as the `fit` method.
+    pub fn fit_vocabulary<T: ToString>(&self, words: &[T]) -> Result<FittedCountVectorizer> {
+        validate_properties(&self.properties)?;
+        let mut vocabulary: HashMap<String, (usize, usize)> = HashMap::with_capacity(words.len());
+        for item in words.iter().map(|w| w.to_string()) {
+            let len = vocabulary.len();
+            // do not care about frequencies/stopwords if a vocabulary is given. Always 1 frequency
+            vocabulary.entry(item).or_insert((len, 1));
+        }
+        let vec_vocabulary = hashmap_to_vocabulary(&mut vocabulary);
+        Ok(FittedCountVectorizer {
+            vocabulary,
+            vec_vocabulary,
+            properties: self.properties.clone(),
+        })
+    }
+}
+
+/// Counts the occurrences of each vocabulary entry, learned during fitting, in a sequence of texts. Each vocabulary entry is mapped
+/// to an integer value that is used to index the count in the result.
+pub struct FittedCountVectorizer {
+    pub(crate) vocabulary: HashMap<String, (usize, usize)>,
+    pub(crate) vec_vocabulary: Vec<String>,
+    pub(crate) properties: VectorizerProperties,
+}
+
+impl FittedCountVectorizer {
+    /// Given a sequence of `n` texts, produces an array of size `(n, vocabulary_entries)` where column `j` of row `i`
+    /// is the number of occurrences of vocabulary entry `j` in the text of index `i`. Vocabulary entry `j` is the string
+    /// at the `j`-th position in the vocabulary.
+    pub fn transform<T: ToString, D: Data<Elem = T>>(
+        &self,
+        x: &ArrayBase<D, Ix1>,
+    ) -> Array2<usize> {
+        let (vectorized, _) = self.get_term_and_document_frequencies(x);
+        vectorized
+    }
+
+    /// Constains all vocabulary entries, in the same order used by the `transform` method.
+    pub fn vocabulary(&self) -> &Vec<String> {
+        &self.vec_vocabulary
+    }
+
+    pub(crate) fn get_term_and_document_frequencies<T: ToString, D: Data<Elem = T>>(
+        &self,
+        x: &ArrayBase<D, Ix1>,
+    ) -> (Array2<usize>, Array1<usize>) {
+        let mut vectorized = Array2::zeros((x.len(), self.vocabulary.len()));
```

Comment:
Yes, that makes sense, as soon as I have some benchmarks ready I will look into that. Perhaps it could be behind a flag? Or maybe we should just say: if you don't want it as a sparse matrix just call the related `sprs` method?